### ISSUE TITLE
multiclass detection has no output, even for training data

### ISSUE THREAD

**Issue by enchantress1016** [matterport/Mask_RCNN#685]
Hi, I'm working on a multi-class detection problem while it's hard to distinguish the items from human vision. I first work on a binary detection (either one class of the item or background) and it works good, though the results couldn't be that accurate (I guess because even from human vision it is hard to tell). 
However, when working with joint multi-class detection problem, I find it normally has nothing detected and there is no output even for training data. Since during the train, the loss and val_loss is not high, I suppose it should at least work good on the training data, is it possible that I mess up with the code? Or it is just impossible to do multi-class detection for the problem.

This is the loss I get after training for multiclass, but I couldn't get any output even for training data in the problem.
Epoch 40/40
100/100 [==============================] - 84s 845ms/step - loss: 0.7847 - rpn_class_loss: 0.0044 - rpn_bbox_loss: 0.3186 - mrcnn_class_loss: 0.1197 - mrcnn_bbox_loss: 0.1222 - mrcnn_mask_loss: 0.2197 - val_loss: 5.7929 - val_rpn_class_loss: 0.0367 - val_rpn_bbox_loss: 5.2939 - val_mrcnn_class_loss: 0.0052 - val_mrcnn_bbox_loss: 0.3068 - val_mrcnn_mask_loss: 0.1503
In comparison, this is the loss I get for training binary problem (one of the classes in the multi problem), and I can get output for every pic in this problem.
100/100 [==============================] - 68s 683ms/step - loss: 0.2497 - rpn_class_loss: 0.0013 - rpn_bbox_loss: 0.0878 - mrcnn_class_loss: 0.0090 - mrcnn_bbox_loss: 0.0417 - mrcnn_mask_loss: 0.1101 - val_loss: 1.8302 - val_rpn_class_loss: 0.0020 - val_rpn_bbox_loss: 0.3195 - val_mrcnn_class_loss: 0.0171 - val_mrcnn_bbox_loss: 0.5660 - val_mrcnn_mask_loss: 0.9256

Could anyone elaborate on the problem? I have no clue where should I start working on. Your help is really appreciate, thank you:)

**Comment by g2-bernotas** [comment]
So all the images are empty? Are you sure you have the correct number of classes? Are you uploading the masks correctly?

**Comment by enchantress1016** [comment]
@g2-bernotas Thank you for the reply. Yes I believe so, I've modified the class number in configuration part and for the label .json file, the number of classes matches my labels. I checked the r["class_ids"] in the detect function and it's empty, so I believe it doesn't get any instance detected.

**Comment by babbu3682** [comment]
@enchantress1016 same problem.... Did you fix that?
